### PR TITLE
[CPM-186] placeholder courses

### PR DESCRIPTION
**Pull request by anian100**
## Changes

- New field added to courses to specify if the course is a placeholder
- New checkbox field when editing/creating courses to specify placeholder course
- Dragging placeholder courses from course pill adds new placeholder course to plan
- Dragging placeholder courses from semester box moves existing placeholder course
- Placeholder course pills are light blue

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-186]

[CPM-186]: https://momentum-agile.atlassian.net/browse/CPM-186

### CODE REVIEW

File: client/src/pages/Courses/editables/IsPlaceholderField.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Text, Checkbox, Tag } from "@chakra-ui/core";
+import { colors as c } from "../../../colors";
+
+const IsPlaceholderField = ({ value, isEditing, onChange }) => {
+    const [isPlaceholder, setIsPlaceholder] = useState([]);
+    console.log(value);
```
Comment (FocalChord) [inline client/src/pages/Courses/editables/IsPlaceholderField.js:7]: remove

File: client/src/pages/Courses/editables/IsPlaceholderField.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    <Flex justify="center">
+                        <Tag height="10px" mt="20px" size="md" rounded="full" variant="solid" variantColor="teal" mr={1}>
+                            Placeholder
+                        </Tag>
+                    </Flex>
+                )
+            ) : (
+                <Flex direction="column" alignItems="center">
+                    <Text textAlign="center" color={c.darkBlue} fontWeight="bold">
+                        Placeholder Course
```
Comment (vee-christella) [inline client/src/pages/Courses/editables/IsPlaceholderField.js:29]: Can you please add a tooltip that the user can hover to understand what a placeholder course really means? So the icon with the (?) that might say:

`A placeholder course can be added multiple times within the same plan.`

File: client/src/pages/Plan/CoursePill.js [hunk 3]
```diff
@@ -15,10 +15,17 @@ const CoursePill = ({ courseName }) => {
...
 
     return (
-        <Box flex="0 0 100%" borderRadius="10px" borderRight="solid white" maxWidth="33%" marginTop="10px" style={{ opacity }}>
-            <Flex p={2} cursor="pointer" width="90%" backgroundColor={c.midnightBlue} borderRadius="8px" ref={drag}>
+        <Box flex="0 0 100%" borderRadius="10px" borderRight="solid white" maxWidth="50%" marginTop="10px" style={{ opacity }}>
+            <Flex
+                p={2}
+                cursor="pointer"
+                width="90%"
+                backgroundColor={isPlaceholder ? c.lightBlue : c.midnightBlue}
```
Comment (vee-christella) [inline client/src/pages/Plan/CoursePill.js:23]: Can we change the colour to `c.uoaBlue`?

```suggestion
                backgroundColor={isPlaceholder ? c.uoaBlue : c.midnightBlue}
```

**Review by vee-christella — COMMENTED** [review]
Everything LGTM, but just wanna add some visual features :)

**Review by Richardng12 — APPROVED** [review]
lgtm after changes

**Review by HongShi10 — APPROVED** [review]
LGTM after changes

**Pull request merged**